rotkehlchen/db/evmtx.py

```diff
@@ -215,21 +227,25 @@ def get_evm_transactions(
         """
         query, bindings = filter_.prepare()
         query, bindings = self._form_evm_transaction_dbquery(query, bindings, has_premium)
-        results = cursor.execute(query, bindings)
+        grouped_transactions: dict[int, tuple[Any, ...]] = {}  # Group results by transaction identifier  # noqa: E501
+        for result in cursor.execute(query, bindings):
+            if (tx_identifier := result[12]) not in grouped_transactions:  # Store base transaction data + empty auth list  # noqa: E501
+                grouped_transactions[tx_identifier] = (*result[:self.AUTHORIZATION_DATA_START_INDEX], [])  # noqa: E501
+
+            if (nonce := result[self.AUTHORIZATION_DATA_START_INDEX]) is not None:  # add authorizations if exist  # noqa: E501
```

Comment:
I asked this in the previous PR

```suggestion
            if (nonce := result[self.AUTHORIZATION_DATA_START_INDEX]) is not None:  # add authorizations if they exist  # noqa: E501
```